source/DSCResources/DSC_CMAssetIntelligencePoint/DSC_CMAssetIntelligencePoint.psm1

```diff
@@ -0,0 +1,446 @@
+                    $setSchedule = $true
+                }
+                elseif (($state.Schedule) -and ($Schedule.RecurCount -ne 0))
+                {
+                    $newSchedule = @{
+                        RecurInterval = $Schedule.RecurInterval
+                        RecurCount    = $Schedule.RecurCount
+                    }
+
+                    $desiredSchedule = New-CMSchedule @newSchedule
+
+                    $currentSchedule = @{
+                        RecurInterval = $state.Schedule.RecurInterval
+                        RecurCount    = $state.Schedule.RecurCount
+                    }
+
+                    $stateSchedule = New-CMSchedule @currentSchedule
+
+                    $array = @('DayDuration','DaySpan','HourDuration','HourSpan','IsGMT')
+
+                    foreach ($item in $array)
+                    {
+                        if ($desiredSchedule.$($item) -ne $stateSchedule.$($item))
+                        {
+                            Write-Verbose -Message ($script:localizedData.ScheduleItem `
+                                -f $item, $($desiredSchedule.$($item)), $($stateSchedule.$($item)))
+                            $setSchedule = $true
+                        }
+                    }
+                }
+
+                if ($setSchedule -eq $true)
+                {
+                    if ($Schedule.RecurCount -eq 0)
+                    {
+                        $desiredSchedule = New-CMSchedule -Nonrecurring
+                    }
+                    elseif (-not ($state.Schedule))
+                    {
+                        $desiredScheduleSet = @{
+                            RecurInterval = $Schedule.RecurInterval
+                            RecurCount    = $Schedule.RecurCount
+                        }
+                        $desiredSchedule = New-CMSchedule @desiredScheduleSet
+                    }
+                    $buildingParams += @{
+                        Schedule = $desiredSchedule
+                    }
+                }
+            }
+
+            if ($buildingParams)
+            {
+                Set-CMAssetIntelligenceSynchronizationPoint @buildingParams
+            }
+        }
+        elseif ($state.Ensure -eq 'Present')
+        {
+            Write-Verbose -Message ($script:localizedData.RemoveAPRole -f $SiteServerName)
+            Remove-CMAssetIntelligenceSynchronizationPoint -SiteSystemServerName $SiteServerName -SiteCode $SiteCode
+        }
+    }
+    catch
+    {
+        throw $_
+    }
+    finally
+    {
+        Set-Location -Path "$env:temp"
+    }
+}
+
+<#
+    .SYNOPSIS
+        This will test the desired state.
+
+    .PARAMETER SiteCode
+        Specifies a site code for the Configuration Manager site that manages the system role for the asset intelligence point.
+
+    .PARAMETER SiteServerName
+        Specifies the Site Server to install or configure the role on.
```

Comment:
needs updated something like:

Specifies the Site Server to install the role on.  If the role is already installed on another server this setting will be ignored.

Reply:
updated and pushed